shared/src/commonTest/kotlin/com/mbta/tid/mbta_app/model/PatternsByStopTest.kt

```diff
@@ -300,4 +300,309 @@ class PatternsByStopTest {
         assertEquals(listOf(alert), patternsByStop.alertsHereFor(0, global))
         assertEquals(listOf(alert), patternsByStop.alertsHereFor(1, global))
     }
+
+    @Test
+    fun `resolveRealtimePatternForDirection splits direction groups into headsigns when needed`() {
```

Comment:
suggestion: I'm having a bit of a hard time understanding this test since there is a lot of data defined & different cases asserted. Consider breaking this down into smaller tests that potentially share relevant data

Reply:
Thanks for suggesting this! I quite like the pattern I ended up with with all the shared data in an object that each case references, I'll try to remember to do that sort of thing going forward when multiple tests share similar data.